xdsl/frontend/pyast/code_generation.py

```diff
@@ -196,6 +196,65 @@ def visit_BinOp(self, node: ast.BinOp):
                 f"which does not overload '{overload_name}'.",
             )
 
+    def visit_Call(self, node: ast.Call):
+        # Resolve function
+        assert isinstance(node.func, ast.Name)
+        func_name = node.func.id
+        for key, value in self.type_converter.globals.items():
+            if key == func_name:
+                source_func = value
+                break
+        else:
+            raise CodeGenerationException(
+                self.file,
+                node.lineno,
+                node.col_offset,
+                f"Function '{func_name}' is not defined in scope.",
```

Comment:
It would be nice to have tests for these, I'm not sure how good coverage is for the existing exceptions but it would be nice to have good coverage going forward.